### DISCUSSION TITLE
Upload ecore model

### DISCUSSION THREAD

**Discussion by volodgo** [eclipse-sirius/sirius-components#3682] [Q&A] [answered]
Hello!
I want to upload the ecore model as in this example to Sirius Web
[https://wiki.eclipse.org/Sirius/Tutorials/Mindstorms](https://wiki.eclipse.org/Sirius/Tutorials/Mindstorms)
I click upload model, but I get an empty model
![image](https://github.com/eclipse-sirius/sirius-web/assets/24474478/ae2bbb96-7044-4de4-89a5-a3d0ad53452d)
Or should I load it on the server side?

**Comment by sbegaudeau** [comment, marked as answer]
Hi,

Ecore is not registered by default as a meta model usable in the resource set. This would allow the creation of Ecore models in any project which is not what we desire today. You just have to register both Ecore and its adapter factory to make it work with something like this:

```
@Configuration
public class EcoreConfiguration {
    @Bean
    public EPackage ecorePackage() {
        return EcorePackage.eINSTANCE;
    }

    @Bean
    public AdapterFactory ecoreAdapterFactory() {
        return new EcoreItemProviderAdapterFactory();
    }
}
```

Regards,

Stephane Begaudeau
